Xamarin.Forms.Platform.Android/AppCompat/NavigationPageRenderer.cs

```diff
@@ -792,20 +792,38 @@ void UpdateMenu()
 				else
 				{
 					IMenuItem menuItem = menu.Add(item.Text);
-					FileImageSource icon = item.Icon;
-					if (!string.IsNullOrEmpty(icon))
-					{
-						Drawable iconDrawable = context.GetFormsDrawable(icon);
-						if (iconDrawable != null)
-							menuItem.SetIcon(iconDrawable);
-					}
+					UpdateDrawableMenuItem(context, menuItem, item);
 					menuItem.SetEnabled(controller.IsEnabled);
 					menuItem.SetShowAsAction(ShowAsAction.Always);
 					menuItem.SetOnMenuItemClickListener(new GenericMenuClickListener(controller.Activate));
 				}
 			}
 		}
 
+		/// <summary>
+		/// For override specify tint color menu or specify drawable menu
+		/// Sample:
+		/// base.UpdateMenuItem(...)
+		/// menuItem.Icon.Mutate();
+		/// menuItem.Icon.SetColorFilter(this.Resources.GetColor([ResourceColor]), Android.Graphics.PorterDuff.Mode.SrcIn);
+		/// 
+		/// Sample with drawable badge count:
+		/// menuItem.SetIcon(SetDrawableIcon(menuItem.Icon)));
+		/// 
+		/// In this fact, toolBarItem is necessary for access to custom property like badge count
+		/// </summary>
+		/// <param name="menuItem"></param>
+		protected virtual void UpdateDrawableMenuItem(Context context, IMenuItem menuItem, ToolbarItem toolBarItem)
+		{
+			FileImageSource icon = toolBarItem.Icon;
+			if (!string.IsNullOrEmpty(icon))
+			{
+				Drawable iconDrawable = context.GetFormsDrawable(icon);
```

Comment:
probably can be disposed after use to clear up GREF

Reply:
in your code, you don't dispose iconDrawable. Is it necessary?

Xamarin.Forms.Platform.Android/Platform.cs
Xamarin.Forms.Platform.Android/CellAdapter.cs

Reply:
what do you mean GREF?

Reply:
Its an android reference handle to a native java object that is created and tracked under the hood. It is quite expensive to keep them hanging around so for short lived objects disposal is ideal. It basically says to Android "okay Im done with this now, you can hand the object off to Java" instead of waiting for the GC to do that for us.

Making sure we don't creep GREFs is becoming increasingly needed. There are plenty of places where we screw this up for sure, as you already pointed out. It's not needed as you ask, but it does help make sure we don't end up reserving many thousands of GREFs and impact performance quite badly.

Reply:
ok, i have test and change code.